### ISSUE TITLE
High CPU usage

### ISSUE THREAD

**Issue by NeonWizard** [continuedev/continue#866]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [x] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://continue.dev/docs/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS: Mac Ventura 13.3.1
- Continue: v0.8.12
- IDE: VSCode
```

### Description

CPU usage goes to extremely high usage when Continue is enabled. Not actively using the plugin features. VSCode began freezing today without having made any changes, and the bisect tool discovered Continue as the culprit. Did not have any issues prior to today.

### To reproduce

1. Use Continue

**Comment by LangLangBart** [comment]
> - [x] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug


A very recent report, #856, suggests disabling indexing in the `$HOME/.continue/config.json` file.

**Comment by sestinj** [comment]
@NeonWizard I will be adding a pause button to indexing as well as trying to run it in a separate process to avoid the slowdown.

I'm going to close this issue so that we can keep all conversation in a single thread, but thank you for emphasizing this problem!

**Comment by denieler** [comment]
In my case I'm having close to 100% CPU usage and it doesn't stop after 10-15-40 minutes, just keeps spinning

**Comment by taaz** [comment]
After installing Continue in vscode, I also really high cpu usages and it seemed to never stop.

After reading this issue and the comments here, I tried **going to Continue in the sidebar**, **clicked on the three dots in the upper right corner** of the sidebar and saw that the codebase indexing was complete. I **used the link "Click to re-index"** and **waited for that indexing to complete**. After that I restarted vscode (don't know if that was necessary or if I could have just waited a moment for the cpu usage to go down). **After the restart, my cpu usage is fine** and the index is used.

Maybe that helps somebody?
